### ISSUE TITLE
Fssm breaks with jruby

### ISSUE THREAD

**Issue by patcheng** [jruby/jruby#344]
copy-n-pasted from https://github.com/ttilley/fssm/issues/38

still seeing it with 1.7.0 RC2 running in 1.9 mode.

```ruby
jruby 1.6 (or 1.7-head) in ruby 1.9 mode:

jruby-1.6.6 :001 > tree = FSSM::Tree::Cache.new
 => #<FSSM::Tree::Cache:0x48a5bc0c @children={}> 
jruby-1.6.6 :002 > tree.set File.expand_path('.')
 => 2012-02-04 18:15:16 -0500 
jruby-1.6.6 :003 > tree.each {|item| puts item[0]}
//
//Volumes
//Volumes/Code
//Volumes/Code/github
//Volumes/Code/github/fssm
jruby-1.6.6 :004 > tree.inject({}) {|hash, (path, node)| hash[path.to_s] = node.mtime; hash}
NoMethodError: undefined method `mtime' for nil:NilClass
    from (irb):4:in `evaluate'
    from org/jruby/RubyProc.java:258:in `call'
    from /Volumes/Code/github/fssm/lib/fssm/tree.rb:39:in `each'
```

For some reason, when in 1.9 mode, jruby doesn't support expanding an array into multiple parts using the |hash, (path, node)| syntax. As a matter of fact, it actually appears to be slicing the array into parameters ahead of time! Even when using a single block parameter for that second object, all I get back is the path (and not the array). If I got back the correct object (the array), then I could at least abandon the syntax and do path, node = array. It doesn't appear that I can fix this in FSSM.

**Comment by MSNexploder** [comment]
this is fixed in 1.7.20 and 9.0.0.0.pre2

**Comment by headius** [comment]
Thanks again!
